node/tessel-export.js

```diff
@@ -557,17 +562,21 @@ Tessel.Pin.prototype.addListener = function(mode, callback) {
       throw new Error(`Interrupts are not supported on pin ${this.pin}. Pins 2, 5, 6, and 7 on either port support interrupts.`);
     }
 
-    if ((mode === 'high' || mode === 'low') && !callback.listener) {
-      throw new Error('Cannot use "on" with level interrupts. You can only use "once".');
-    }
-
-    if (this.interruptMode !== mode) {
+    // For one-time 'low' or 'high' event
+    if (mode === 'low' || mode === 'high') {
+      if (!callback.listener) {
+        throw new Error('Cannot use "on" with level interrupts. You can only use "once".');
+      }
+      // Can't set multiple listeners when using 'low' or 'high'
```

Comment:
I think I have it sorted out? The original purpose of the restructure was the result of there now being two different errors to check for... one is related to trying to set a `.on()` listener for 'high' or 'low' and one is related to trying to add a 'high' or 'low' listener when there is already a 'change' listener.

The problem, however, was that it only caught it if the 'change' listener was added before a 'high' or 'low' listener.

So I modified the code (and also addressed your second question) with this:

```javascript
Tessel.Pin.prototype.addListener = function(mode, callback) {
  if (typeof Tessel.Pin.interruptModes[mode] !== 'undefined') {
    if (!this.interruptSupported) {
      throw new Error(`Interrupts are not supported on pin ${this.pin}. Pins 2, 5, 6, and 7 on either port support interrupts.`);
    }

    // For one-time 'low' or 'high' event
    if ((mode === 'low' || mode === 'high') && !callback.listener) {
      throw new Error('Cannot use "on" with level interrupts. You can only use "once".');
    }

    // Can't set multiple listeners when using 'low' or 'high'
    if (this.interruptMode) {
      if ((mode === 'low' || mode === 'high') || (this.listeners('low')[0] || this.listeners('high')[0])) {
        throw new Error(`Cannot set pin interrupt mode to ${mode}; already listening for ${this.interruptMode}. Can only set multiple listeners with change, rise & fall`);
      }
    }

    // Set the socket reference so the script doesn't exit
    this._port.ref();
    this._setInterruptMode(mode);
  }

  // Add the event listener
  Tessel.Pin.super_.prototype.on.call(this, mode, callback);
};
```
Also made a change to `_setInterruptMode()` so that `rise` & `fall` are actually just considered `change`.
```javascript
Tessel.Pin.prototype._setInterruptMode = function(mode) {
  // rise and fall events will be emitted by change event
  (mode === 'rise' || mode === 'fall') ? this.interruptMode = 'change' : this.interruptMode = mode;
  var bits = mode ? Tessel.Pin.interruptModes[mode] << 4 : 0;
  this._port._simple_cmd([CMD.GPIO_INT, this.pin | bits]);
};
```